src/components/app/MenuButtons/Publish.js

```diff
@@ -120,6 +128,24 @@ class MenuButtonsPublishImpl extends React.PureComponent<PublishProps> {
     );
   }
 
+  static getDerivedStateFromProps(
+    props: PublishProps,
+    state: PublishState
+  ): $Shape<PublishState> | null {
+    if (state.prevCompressedPromise !== props.compressedProfileBlobPromise) {
+      return {
+        // Invalidate the old error info
+        prevCompressedPromise: props.compressedProfileBlobPromise,
+        compressError: null,
+      };
+    }
+    return null;
+  }
+
+  _onCompressError = (error) => {
+    this.setState({ compressError: error });
+  };
```

Comment:
I'm not sure this is the best approach, but I wanted to avoid copying the "unwrap the promise" thing in both components.